cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -216,6 +216,95 @@ public List<LocationGroup> getLocationGroups(@Nullable String officeId,
+     * @Param officeId The office id to use for the query.
+     * @Param locCategoryLike A regex to use to filter the location categories.  May be null.
+     * @Param sharedRefLocLike A where clause to filter the shared_loc_alias_id.  May be null.
+     * @return A list of all location groups for the given parameters.
+     */
+
+    public List<LocationGroup> getLocationGroups(String officeId, String locCategoryLike, String sharedRefLocLike) {
+        AV_LOC_GRP_ASSGN alga = AV_LOC_GRP_ASSGN.AV_LOC_GRP_ASSGN;
+        AV_LOC_CAT_GRP alcg = AV_LOC_CAT_GRP.AV_LOC_CAT_GRP;
+
+        final RecordMapper<Record, Pair<LocationGroup, AssignedLocation>> mapper = grpRecord -> {
+            LocationCategory category = buildLocationCategory(grpRecord);
+
+            LocationGroup group = buildLocationGroup(grpRecord, category);
+            AssignedLocation loc = buildAssignedLocation(grpRecord);
+
+            return new Pair<>(group, loc);
+        };
+
+        Map<LocationGroup, List<AssignedLocation>> map = new LinkedHashMap<>();
+
+        SelectConnectByStep<? extends Record> connectBy;
+        SelectOnConditionStep<? extends Record> onStep = dsl.select(
+                        alcg.CAT_DB_OFFICE_ID,
+                        alcg.LOC_CATEGORY_ID,
+                        alcg.LOC_CATEGORY_DESC,
+                        alcg.GRP_DB_OFFICE_ID,
+                        alcg.LOC_GROUP_ID,
+                        alcg.LOC_GROUP_DESC,
+                        alcg.LOC_GROUP_ATTRIBUTE,
+                        alcg.SHARED_LOC_ALIAS_ID,
+                        alcg.SHARED_REF_LOCATION_ID,
+                        alga.DB_OFFICE_ID,
+                        alga.LOCATION_ID,
+                        alga.ALIAS_ID,
+                        alga.ATTRIBUTE,
+                        alga.REF_LOCATION_ID)
+                .from(alcg).leftJoin(alga)
+                .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
+                        .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
+
+
+        Condition condition = noCondition();
+        if (locCategoryLike != null && !locCategoryLike.isEmpty()) {
+            condition = caseInsensitiveLikeRegex(alcg.LOC_CATEGORY_ID, locCategoryLike);
+        }
+
+        if (sharedRefLocLike != null && !sharedRefLocLike.isEmpty()) {
+            condition = condition.and(caseInsensitiveLikeRegex(alcg.SHARED_REF_LOCATION_ID, sharedRefLocLike));
+        }
+
+        if (officeId != null) {
+            if (CWMS.equalsIgnoreCase(officeId)) {
+                connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.eq(CWMS)
+                        .and(alcg.GRP_DB_OFFICE_ID.eq(CWMS))
+                        .and(condition)
+                );
+            } else {
+                connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.in(CWMS, officeId)
+                        .and(alcg.GRP_DB_OFFICE_ID.in(CWMS, officeId))
+                        .and(alga.DB_OFFICE_ID.isNull().or(alga.DB_OFFICE_ID.eq(officeId)))
```

Comment:
you're looking specifically for location groups so you don't want null ones

Reply:
also, all of these checks should ignore case (by calling upper on both sides of the `eq`)